contracts/EverlongStrategy.sol

```diff
@@ -0,0 +1,482 @@
+    bool public immutable asBase;
+
+    /// @dev The Hyperdrive's PoolConfig.
+    IHyperdrive.PoolConfig internal _poolConfig;
+
+    // ╭───────────────────────────────────────────────────────────────────────╮
+    // │                                 STATE                                 │
+    // ╰───────────────────────────────────────────────────────────────────────╯
+
+    /// @dev Configuration for how `_tend(..)` is performed.
+    IEverlongStrategy.TendConfig internal _tendConfig;
+
+    /// @dev Structure to store and account for everlong-controlled positions.
+    Portfolio.State internal _portfolio;
+
+    // ╭───────────────────────────────────────────────────────────────────────╮
+    // │                              CONSTRUCTOR                              │
+    // ╰───────────────────────────────────────────────────────────────────────╯
+
+    /// @notice Creates a new EverlongStrategy.
+    /// @param _asset Asset to use for the strategy.
+    /// @param __name Name for the strategy.
+    /// @param _hyperdrive Address of the Hyperdrive instance.
+    /// @param _asBase Whether `_asset` is Hyperdrive's base asset.
+    constructor(
+        address _asset,
+        string memory __name,
+        address _hyperdrive,
+        bool _asBase
+    ) BaseStrategy(_asset, __name) {
+        hyperdrive = _hyperdrive;
+        asBase = _asBase;
+        _poolConfig = IHyperdrive(_hyperdrive).getPoolConfig();
+    }
+
+    // ╭───────────────────────────────────────────────────────────────────────╮
+    // │                          STRATEGY OVERRIDES                           │
+    // ╰───────────────────────────────────────────────────────────────────────╯
+
+    /// @inheritdoc BaseStrategy
+    function _deployFunds(uint256) internal override {
+        // Do nothing.
+        // Opening longs on Hyperdrive is sandwichable so funds should only be
+        // deployed when the `keeper` calls `tend()`.
+        return;
+    }
+
+    /// @inheritdoc BaseStrategy
+    function _freeFunds(uint256 _amount) internal override {
+        // Close all matured positions (if any).
+        // TODO: Determine whether `_tendConfig.positionClosureLimit` should
+        //       affect this.
+        uint256 output = _closeMaturedPositions(0);
+
+        // Close immature positions if additional funds need to be freed.
+        if (_amount > output) {
+            _closePositions(_amount - output);
+        }
+    }
+
+    /// @inheritdoc BaseStrategy
+    function _harvestAndReport()
+        internal
+        override
+        returns (uint256 _totalAssets)
+    {
+        // If the strategy isn't shut down, call `_tend()` to close mature
+        // positions and spend idle if needed.
+        if (!TokenizedStrategy.isShutdown() && _tendTrigger()) {
+            _tend(ERC20(asset).balanceOf(address(this)));
+        }
+
+        // Recalculate the value of assets the strategy controls.
+        _totalAssets = calculateTotalAssets();
```

Comment:
You mentioned that we calculate total assets after tend to avoid reporting losses. That would be a good piece of documentation to add here.

Reply:
Ok gotcha.